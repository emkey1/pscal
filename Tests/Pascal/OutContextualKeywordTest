program OutContextualKeywordTest;

procedure Increment(out value: integer);
begin
  value := value + 1;
end;

procedure ShowOutIdentifier;
var
  out: integer;
begin
  out := 10;
  writeln('local out identifier=', out);
end;

var
  outValue: integer;
  out: integer;

begin
  outValue := 41;
  Increment(outValue);
  writeln('after Increment: outValue=', outValue);

  out := 3;
  writeln('global out identifier=', out);

  ShowOutIdentifier;
end.
