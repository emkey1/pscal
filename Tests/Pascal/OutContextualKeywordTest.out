after Increment: outValue=42
global out identifier=3
local out identifier=10
